tests/test_rfn_query.py

```diff
@@ -0,0 +1,138 @@
+import datetime as dt
+import pandas as pd
+from pandas.compat import lrange
+from pandas import DataFrame
+
+from bgbb.sql.sql_utils import run_rec_freq_spk, to_s3_fmt
+import pyspark
+from pytest import fixture
+
+
+spark = pyspark.sql.SparkSession.builder.appName("test").getOrCreate()
+
+
+def int2date(i):
+    return model_start_date + dt.timedelta(days=i - 1)
+
+
+model_start_date = dt.date(2018, 6, 1)
+MAX_OPPORTUNITIES = 9
+DURATION = 10
+rfn_cols = ["Recency", "Frequency", "N"]
+
+
+def generate_clients_daily():
```

Comment:
```suggestion
@fixture
def generate_clients_daily():
```

The decorator is common when writing pytest fixtures.

Reply:
I was doing a bunch of things like

```
@fixture
def rfn_df():
    rfn_df = do_stuff()
    do_more_stuff(rfn_df)
    return rfn_df
```

where (since it's a fixture), the variable name and fixture/function name were the same. It seemed like a bad pattern to reuse the name like that, so I figured having a separate name for the generating function and fixture would be better.

Guess that's not recommended?

Reply:
Yeah, namespace clashes can make things unclear. You could come up with an alternative name for the fixture like `test_activity_df`, or try to avoid using a lot of intermediate state.